src/components/ColorPickerInput/utils.ts

```diff
@@ -0,0 +1,16 @@
+export const hexRegexp = /^#[a-fA-F0-9]{6}$/;
```

Comment:
All regex's in your code are used with global flag. Let's add it to all the regulars and use it like this:

```
export const hexRegexp = /^#[a-fA-F0-9]{6}$/g;

new RegExp(hexRegexp, 'g').test(value) -> hexRegexp.test(value)
```

Reply:
it not working properly, check  https://learn.javascript.ru/regexp-methods

P.S. If i use  `regexp.test` more than one it's get wrong result